src/main/java/sx/blah/discord/util/MessageHistoryRange.java

```diff
@@ -0,0 +1,280 @@
+/*
+ *     This file is part of Discord4J.
+ *
+ *     Discord4J is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     Discord4J is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Lesser General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Lesser General Public License
+ *     along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package sx.blah.discord.util;
+
+import sx.blah.discord.handle.impl.obj.Channel;
+import sx.blah.discord.handle.obj.IChannel;
+import sx.blah.discord.handle.obj.IMessage;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class represents a range of a channel's message history. It includes a start and endpoint to clearly define the
+ * range of messages. These endpoints can be either timestamps or messages, and can be inclusive or exclusive. This
+ * class also contains some logic regarding getting the first message of the range (the message closest to `start` that
+ * is in range).
+ * It is used in {@link MessageHistoryBuilder} and {@link MessageHistoryIterator} to define the range being requested.
+ *
+ * @see MessageHistoryBuilder
+ * @see MessageHistoryIterator
+ * @see Point
+ */
+public class MessageHistoryRange {
+
+	private final Channel channel;
+
+	private final Point start;
+	private final Point end;
+
+	private final boolean chronological;
+
+	public MessageHistoryRange(IChannel channel, Point start, Point end) {
```

Comment:
It's more an issue of the design of the class- I wanted to move away from "before" and "after" type setup for clarity, which is why I designed it as "startAt" and "endAt". Do you have any suggestions for how this could be redesigned to allow for that?